src/tagstudio/qt/flowlayout.py

```diff
@@ -5,71 +5,85 @@
 
 """PySide6 port of the widgets/layouts/flowlayout example from Qt v6.x."""
 
+from typing import Literal, override
+
 from PySide6.QtCore import QMargins, QPoint, QRect, QSize, Qt
-from PySide6.QtWidgets import QLayout, QSizePolicy, QWidget
+from PySide6.QtWidgets import QLayout, QLayoutItem, QSizePolicy, QWidget
+
+IGNORE_SIZE = "ignore_size"
 
 
 class FlowWidget(QWidget):
-    def __init__(self, parent=None) -> None:
+    def __init__(self, parent: QWidget | None = None) -> None:
         super().__init__(parent)
-        self.ignore_size: bool = False
+        self.setProperty(IGNORE_SIZE, False)  # noqa: FBT003
 
 
 class FlowLayout(QLayout):
-    def __init__(self, parent=None):
+    def __init__(self, parent: QWidget | None = None) -> None:
         super().__init__(parent)
 
         if parent is not None:
             self.setContentsMargins(QMargins(0, 0, 0, 0))
 
-        self._item_list = []
+        self._item_list: list[QLayoutItem] = []
         self.grid_efficiency = False
 
     def __del__(self):
         item = self.takeAt(0)
         while item:
             item = self.takeAt(0)
 
-    def addItem(self, item):  # noqa: N802
-        self._item_list.append(item)
+    @override
+    def addItem(self, arg__1: QLayoutItem) -> None:
+        self._item_list.append(arg__1)
 
-    def count(self):
+    @override
+    def count(self) -> int:
         return len(self._item_list)
 
-    def itemAt(self, index):  # noqa: N802
-        if 0 <= index < len(self._item_list):
-            return self._item_list[index]
+    @override
+    def itemAt(self, arg__1: int) -> QLayoutItem | None:
+        if 0 <= arg__1 < len(self._item_list):
+            return self._item_list[arg__1]
```

Comment:
According to pyright the parent method's argument is called `index` and the pyright comment should take care of the complaint.
```suggestion
    @override
    def itemAt(self, index: int) -> QLayoutItem | None:  # pyright: ignore[reportIncompatibleMethodOverride]
        if 0 <= index< len(self._item_list):
            return self._item_list[index]
```